robottelo/rerun_failures/uncollector.py

```diff
@@ -0,0 +1,111 @@
+import logging
+
+import pytest
+
+from robottelo.config import settings
+from robottelo.report_portal.portal import ReportPortal
+
+LOGGER = logging.getLogger(__name__)
+
+
+def _get_launch(rp):
+    """Returns the latest launch of satellite"""
+    _version = settings.server.version
+    sat_version = f'{_version.base_version}.{_version.epoch}'
+    launch = next(iter(rp.launches(sat_version=sat_version).values()))
+    # Stop session based on RP Launch statistics and info
+    pass_percent = round(
+        100 * (int(launch.statistics['passed']) / int(launch.statistics['total']))
+    )
+    if launch.info['isProcessing']:
+        pytest.skip(f'The latest launch of satellite version {sat_version} is not Finished yet.')
+    if pass_percent < 80:
+        pytest.skip(
+            f'The latest Satellite launch has only {pass_percent}% tests passed. '
+            'Which is lesser than the threshold of 80%. '
+            'Examine the failures thoroughly and check for any major issue.'
+        )
+    return launch
+
+
+def _get_failed_tests(launch, fail_args):
+    """Returns failed test names from Report Portal Launch based on arguments"""
+    tests = []
+    if fail_args == ['all']:
+        test_names = [*launch.tests(status='failed').keys()]
```

Comment:
Why include `*` here? simply wrapping the keys iterator should cast it to a list.